src/ast/passes/semantic_analyser.cpp

```diff
@@ -401,8 +401,11 @@ void SemanticAnalyser::visit(Builtin &builtin)
 
     ProbeType type = single_provider_type();
 
-    if (type == ProbeType::kfunc || type == ProbeType::kretfunc ||
-        type == ProbeType::uprobe) {
+    if (type == ProbeType::invalid) {
+      LOG(ERROR, builtin.loc, err_) << "The args builtin must be used in "
+                                       "probes with a single probe type";
```

Comment:
How about this instead:

```suggestion
      LOG(ERROR, builtin.loc, err_) << "The args builtin can only be used in "
                                       "actions associated with a single probe type. To fix this simply create multiple action blocks; one for each probe.";
```

Reply:
We don't use the word "action" nearly at all (except for the manual) and use "probe" everywhere instead. That being said, I'm not strictly against changing this (although I have a slight preference for "probe") but I'd argue that we should stay consistent for now.

Reply:
I don't want to get into lengthy discussions on error messages but I'd prefer keeping them compact. What about just mentioning the actual probe types? That should give enough hint:
`The args builtin can only be used within the context of a single probe type (have "kfunc", "tracepoint").`

Reply:
> I don't want to get into lengthy discussions on error messages

I'm trying not to honestly, but it's difficult 😄  TBH, I don't understand the `(have "kfunc", "tracepoint")` part, what does this mean? Also, why not mention the solution to this issue if it's easy enough to display to users? How about this as a more compact version of what I suggested above:
`The args builtin can only be used within the context of a single probe type e.g. "probe1 {}" is valid while "probe1,probe2 {}" is not.`